semantic_conventions/trace/general.yaml

```
@@ -106,7 +106,7 @@ groups:
         type: int
         brief: 'Local socket port number.'
         requirement_level:
-          recommended: If defined for the address family and if different than `net.host.port` and if `net.sock.host.addr` is set.
+          conditionally_required: If defined for the address family and if different than `net.host.port` and if `net.sock.host.addr` is set.
```

This now makes it implicitly "opt-in" in all other cases. I think the condition text should be amended such as

```suggestion
          conditionally_required: If defined for the address family and if different than `net.host.port` and if `net.sock.host.addr` is set. In other cases, it is still recommended to set this.
```